Library/Homebrew/cmd/audit.rb

```diff
@@ -676,6 +676,10 @@ def audit_text
     if text =~ /def plist/ && text !~ /plist_options/
       problem "Please set plist_options when using a formula-defined plist."
     end
+
+    if text =~ %r{require "language/go"} && text !~ /go_resource/
```

Comment:
Well, `text` is not a `String` instance (like in your `irb` example), but a `FormulaText` instance as you can learn from the backtrace. Unfortunately, the latter doesn't exactly behave like a plain string (though it implements `=~` and thus gives a good first impression), so my suggestion to use `include?` is somewhat moot …